packages/styled-components/src/utils/hyphenateStyleName.js

```diff
@@ -27,7 +26,7 @@ const prefixAndLowerCase = (char: string): string => `-${char.toLowerCase()}`;
  * @return {string}
  */
 export default function hyphenateStyleName(string: string): string {
-  return uppercaseCheck.test(string)
+  return uppercasePattern.test(string)
```

Comment:
I suppose this is an accidental change? This purposefully uses a non-global regex to avoid the `lastIndex` being moved around. So let's restore `uppercaseCheck` here and only change `uppercasePattern` to a literal :)

Reply:
Oh I see now! :) Amended fix resulting in 813329a.